src/header/types/accept.rs

```diff
@@ -0,0 +1,325 @@
+    where   
+        Self: Sized
+    {
+        &HeaderName::Accept
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MediaType{
+    m_type: Mime,
+    quality: Option<QualityParam>
+
+}
+
+impl Default for MediaType {
+
+    fn default() -> Self {
+        MediaType{
+            m_type: mime::TEXT_PLAIN,
+            quality: None
+        }
+    }
+}
+
+impl Eq for MediaType {}
+
+impl Hash for MediaType {
+
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.m_type.hash(state);
+    }
+}
+
+impl MediaType{
+
+    pub fn new(m_type: Mime, quality: Option<QualityParam>) -> Self {
+        MediaType{
+            m_type,
+            quality
+        }
+    }
+
+    /// Returns a `&str` representation of the media type.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::convert::TryFrom;
+    ///
+    /// use rtsp::header::types::accept::MediaType;
+    /// use rtsp::header::types::accept::QualityParam;
+    /// use mime::*;
+    /// use std::str::FromStr;
+    ///
+    /// assert_eq!(MediaType::new(mime::STAR_STAR, None).as_str(), "*/*");
+    /// assert_eq!(MediaType::new(Mime::from_str("video/*").unwrap(), Some(QualityParam::new(0.5))).as_str(), "video/* ;q=0.5");
+    /// ```
+    pub fn as_str(&self) -> String {
+        let m_type = self.m_type.to_string();
+        match &self.quality {
+            Some(quality) => format!("{} ;{}", m_type, quality.as_str()),
+            None => format!("{}", m_type)
+        }
+    }
+}
+
+impl<'accept> TryFrom<&'accept [u8]> for MediaType {
+    type Error = AcceptError;
+
+    fn try_from(value: &'accept [u8]) -> Result<Self, Self::Error> {
+        let mut split = value.split(|element| element.clone() == ';' as u8);
+        let mut decoded = MediaType::default();
+        if let Some(mediatype) = split.next() {
+            if let Ok(mtype) = str::from_utf8(mediatype).unwrap().parse() {
+                decoded.m_type = mtype;
+                if let Some(quality) = split.next() {
+                    if let Ok(quality_param) = QualityParam::try_from(quality) {
+                        decoded.quality = Some(quality_param);
+                    } else {
+                        return Err(AcceptError("unable to parse quality parameter"))
+                    }
+                } 
+            } else{
+                return Err(AcceptError("unable to parse mime type"))
+            }
+        }
+        Ok(decoded)
+    }
+}
+
+impl<'accept> TryFrom<&'accept str> for MediaType {
+    type Error = AcceptError;
+
+    fn try_from(value: &'accept str) -> Result<Self, Self::Error> {
+        let bytes = value.as_bytes();
+        Self::try_from(bytes)
+    } 
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct QualityParam(f32);
+
+impl<'quality> TryFrom<&'quality [u8]> for QualityParam {
+    type Error = AcceptError;
+
+    fn try_from(value: &'quality [u8]) -> Result<Self, Self::Error> {
+        let stringify = String::from_utf8(value.to_vec()).unwrap();
```

Comment:
Same here, we shouldn't assume it's valid UTF-8. Also prefer using `str::from_utf8(value)` as it doesn't allocate.